webplugin/js/app/mck-app.js

```diff
@@ -311,6 +317,82 @@ function ApplozicSidebox() {
             return false;
         }
     }
+
+    function loadResourceAsync(src) {
+        return new Promise((resolve, reject) => {
+            let element;
+
+            if (src.endsWith('.js')) {
+                element = document.createElement('script');
+                element.src = src;
+                element.type = 'text/javascript';
+            } else if (src.endsWith('.css')) {
+                element = document.createElement('link');
+                element.href = src;
+                element.type = 'text/css';
+                element.rel = 'stylesheet';
+            } else {
+                reject(new Error('Unsupported resource type'));
+                return;
+            }
+
+            element.onload = () => resolve();
+            element.onerror = () =>
+                reject(new Error(`Resource load error for ${src}`));
+
+            document.head.appendChild(element);
+        });
+    }
+
+    async function loadFileBasedOnProp(apiData, options) {
+        try {
+            const promises = [];
+
+            if (options.zendeskChatSdkKey) {
+                promises.push(
+                    loadResourceAsync(THIRD_PARTY_SCRIPTS.zendesk.js)
+                );
+            }
+
+            if (
+                Array.isArray(options.preLeadCollection) ||
+                (apiData.collectLead && apiData.leadCollection)
+            ) {
+                const leadCollection =
+                    options.preLeadCollection || apiData.leadCollection;
+
+                const shouldLoadIntl = leadCollection?.some(
+                    (obj) => obj.enableCountryCode
+                );
+
+                if (shouldLoadIntl) {
+                    Object.values(THIRD_PARTY_SCRIPTS.intlForPreChat).forEach(
+                        async (file) => {
+                            promises.push(loadResourceAsync(file));
+                        }
+                    );
+                }
+            }
+
+            if (options.voiceNote) {
+                promises.push(
+                    loadResourceAsync(THIRD_PARTY_SCRIPTS.recorder.js)
```

Comment:
if recorder and frvoice are both required together then better to have a single combined minified file for them

Reply:
That is good suggestion it will not make the extra call

Reply:
done